test/d2i_test.c

```diff
@@ -69,36 +69,37 @@ static int test_bad_asn1()
      * decoder is called.
      */
     len = BIO_read(bio, buf, sizeof buf);
-    if (len < 0)
+    if (!TEST_int_ge(len, 0))
         goto err;
 
     value = ASN1_item_d2i(NULL, &buf_ptr, len, item_type);
     if (value == NULL) {
-        if (expected_error == ASN1_DECODE)
+        if (TEST_int_eq((int)expected_error, (int)ASN1_DECODE))
```

Comment:
Probably not.  I'll remove them.
I'm sure I met a compiler a long time ago that complained about this :(